### PR TITLE
Fast-SCNN

### PR DESCRIPTION
**Pull request by ytian8**
Add Fast SCNN model from Fast-SCNN: Fast Semantic Segmentation Network - BMVC 2019.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# this is debug branch
```
Comment (bryanyzhu) [inline README.md:1]: Remove this line

File: gluoncv/model_zoo/fastscnn.py [hunk 2]
```diff
@@ -0,0 +1,344 @@
+
+"""Fast-SCNN, implemented in Gluon."""
```
Comment (bryanyzhu) [inline gluoncv/model_zoo/fastscnn.py:2]: If most of code from fast-seg, you should write one sentence like "Code adapted from xxxx".

File: gluoncv/model_zoo/fastscnn.py [hunk 3]
```diff
@@ -0,0 +1,344 @@
+
```
Comment (bryanyzhu) [inline gluoncv/model_zoo/fastscnn.py:1]: Remove the first empty line.

File: gluoncv/model_zoo/fastscnn.py [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+        Parameters
+        ----------
+        nclass : int
+            Number of categories for the training dataset.
+        aux : bool
+            Auxiliary loss.
+
+        Reference:
+            Fast-SCNN: Fast Semantic Segmentation Network - BMVC 2019
```
Comment (bryanyzhu) [inline gluoncv/model_zoo/fastscnn.py:22]: Add an arxiv link will be better

File: gluoncv/model_zoo/fastscnn.py [hunk 5]
```diff
@@ -0,0 +1,344 @@
...
+
+    # pylint: disable=arguments-differ
+    def hybrid_forward(self, F, x):
+        return self.block(x)
+
+
+class _ConvBNReLU(HybridBlock):
+    """Conv-BN-ReLU"""
+
+    def __init__(self, in_channels, out_channels, kernel_size=3, stride=1, padding=0, **kwargs):   # check stride 1/2?
```
Comment (bryanyzhu) [inline gluoncv/model_zoo/fastscnn.py:142]: Remove the comment "# check stride 1/2?"

File: gluoncv/model_zoo/fastscnn.py [hunk 6]
```diff
@@ -0,0 +1,344 @@
...
+    }
+    from ..data import datasets
+
+    model = FastSCNN(datasets[dataset].NUM_CLASS, ctx=ctx, **kwargs)
+    model.classes = datasets[dataset].classes
+    model.initialize(ctx=ctx)
+    return model
+
+
+if __name__ == '__main__':
```
Comment (bryanyzhu) [inline gluoncv/model_zoo/fastscnn.py:336]: Remove this main function after you finish debugging.

File: gluoncv/model_zoo/fastscnn.py [hunk 7]
```diff
@@ -0,0 +1,344 @@
...
+        x = F.contrib.BilinearResize2D(x, **self._up_kwargs)
+        return x
+
+    def evaluate(self, x):
+        """evaluating network with inputs and targets"""
+        return self.forward(x)[0]
+
+
+class FeatureFusionModule(HybridBlock):
+    def __init__(self, highter_in_channels, lower_in_channels, out_channels, height, width, scale_factor=4, **kwargs):
```
Comment (bryanyzhu) [inline gluoncv/model_zoo/fastscnn.py:80]: You need `norm_layer=nn.BatchNorm, norm_kwargs=None` here in the '__inti__' function, so that you can use syncbn afterwards. Change line 88 to `norm_layer` later.

**Comment by mli** [comment]
Job PR-1278-12 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-1278/12/index.html 
Code coverage of this PR:  vs. Master:
